cmd/oras/repository/tags.go

```diff
@@ -78,14 +78,36 @@ func showTags(opts showTagsOptions) error {
 	if err != nil {
 		return err
 	}
+	filter := ""
 	if opts.Reference != "" {
-		return fmt.Errorf("unexpected tag or digest %q found in repository reference %q", opts.Reference, opts.RawReference)
+		if isDigestTag(opts.Reference) {
+			filter = opts.Reference
+		} else {
+			desc, err := finder.Resolve(ctx, opts.Reference)
+			if err != nil {
+				return err
+			}
+			filter = desc.Digest.String()
+			fmt.Println(filter)
```

Comment:
Should still return `unexpected tag` error for this code path. Return digest for a tag breaks UX consistency.

Reply:
I'm willing to add the filter as a flag back, but I think this way is better for the user.

Reply:
There are two kinds of features we are discussing about:
1) filtering: based on a provided tag or digest, give the tag result
2) displaying: display corresponding digests instead of tags

Let's keep this PR to the first feature. So we have below use cases:
1.1)  based on provided tag, give the matching tag list
1.2)  based on the provided digest, give the matching tag list

With out feature 2, 1.1 makes no sense since the output will always be the input tag, not a list. Making it returns digest will expand this issue's scope to #806.

Reply:
1.1 makes sense if you want to know what release is `latest` which is exactly what I want to do.

Reply:
In that case shouldn't `oras manifest get --descriptor` be used?

Reply:
If the output is **tags having same digest to the input tag**, that makes sense.

Reply:
I just realize that this is the current implementation, sorry for the gap. Does below UX looks good to you?
- when input is a digest, print all tags associated with the input digest
- when input is a tag, resolve its digest, print all tags associated with the resolved digest

Reply:
> I just realize that this is the current implementation, sorry for the gap. Does below UX looks good to you?
> 
> * when input is a digest, print all tags associated with the input digest
> * when input is a tag, resolve its digest, print all tags associated with the resolved digest

Yes!